lib/commands/skill/add-locales/helper.js

```diff
@@ -0,0 +1,148 @@
+                    Manifest.getInstance().setPublishingLocale(k, { name: 'please change' });
+                }
+            });
+            Manifest.getInstance().write();
+            callback(undefined, iModelSourceByLocales);
+        });
+    });
+}
+
+/**
+ * Based on user's selected locales, and the local project's iModel files, decide where to get the
+ * iModel file to add in user's local project. Exclude reusableLocales themselves from the result.
+ *
+ * @param {Array} selectedLocales
+ * @param {String} profile
+ * @param {Boolean} doDebug
+ * @returns {Function} callback Map { locale: filePath/remoteURI }
+ */
+function _getNewLocaleModelUri(selectedLocales, profile, doDebug, callback) {
+    const skillMetaController = new SkillMetadataController({ profile, doDebug });
+    const localIModelByLocale = skillMetaController.getInteractionModelLocales();
+    const result = new Map();
+    // get from map of templates
+    httpClient.request({
+        url: CONSTANTS.TEMPLATES.INTERACTION_MODEL_MAP,
+        method: CONSTANTS.HTTP_REQUEST.VERB.GET
+    }, 'GET_INTERACTION_MODEL_MAP', doDebug, (error, templateMapResponse) => {
+        if (error || !templateMapResponse.statusCode || templateMapResponse.statusCode !== 200) {
+            return callback('Failed to retrieve the template list. Please run again with --debug to check more details.');
+        }
+        let templateIndexMap = templateMapResponse.body;
+        if (typeof templateMapResponse.body === 'string') {
+            templateIndexMap = JSON.parse(templateMapResponse.body);
+        }
+        // assign for each locale
+        for (const locale of selectedLocales) {
+            if (R.keys(localIModelByLocale).includes(locale)) {
+                continue;
+            }
+            const languageExtractor = (s) => R.slice(0, R.lastIndexOf('-', s), s);
+            const language = languageExtractor(locale);
+            const reusableLocale = R.find((k) => languageExtractor(k) === language)(R.keys(localIModelByLocale));
+            if (reusableLocale) {
+                result.set(locale, localIModelByLocale[reusableLocale]);
+            } else {
+                result.set(locale, templateIndexMap.INTERACTION_MODEL_BY_LANGUAGE[language]);
+            }
+        }
+        callback(undefined, result);
+    });
+}
+
+/**
+ * Retrieve the template content from the url link.
+ * @param {Set} templateSet Set of template URLs to get the content for
+ * @param {Boolean} doDebug
+ * @param {Function} callback Map { templateUrl: bodyContent }
+ */
+function _retrieveTemplatesByLanguage(templateSet, doDebug, callback) {
+    const result = new Map();
+    async.forEach(templateSet.values(), (templateUrl, loopCallback) => {
+        httpClient.request({
+            url: templateUrl,
+            method: CONSTANTS.HTTP_REQUEST.VERB.GET
+        }, 'GET_INTERACTION_MODEL_TEMPLATE', doDebug, (err, response) => {
+            if (err || response.statusCode > 300) {
```

Comment:
above you are also checking for not having status code. As mentioned probably should be handled in http client for consistency.

Reply:
yes move the check of statusCode into httpClient